exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/AsyncPageReader.java

```diff
@@ -83,21 +80,18 @@
 class AsyncPageReader extends PageReader {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(AsyncPageReader.class);
 
-  private ExecutorService threadPool;
-  private long queueSize;
-  private LinkedBlockingQueue<ReadStatus> pageQueue;
-  private ConcurrentLinkedQueue<Future<Void>> asyncPageRead;
-  private long totalPageValuesRead = 0;
-  private Object pageQueueSyncronize = new Object(); // Object to use to synchronize access to the page Queue.
-                                                     // FindBugs complains if we synchronize on a Concurrent Queue
+  private final ExecutableTasksLatch<AsyncPageReaderTask> executableTasksLatch;
```

Comment:
The ExecutableTasksLatch replaces the ExecutorService for concurrent task manipulation:
- Is this the recommended way moving forward?
- There is no javadoc for the ExecutableTasksLatch; you need to document this new class APIs and execution semantics. This is especially true if we want to encourage the use of this implementation pattern.

Reply:
`ExecutableTasksLatch` encapsulates `ExecutorService` for parallel tasks that require a synchronization point at cancellation. For all other tasks `ExecutorService` can be used directly.

I'll add more documentation after the first round of review.

Reply:
The documentation would be useful for the reviewer too. It is easier to check the code if one knows what what guarantees ExecutableTasksLatch is supposed to provide.

Reply:
Note that the class comment for AsyncPageReader needs to be updated too.

Reply:
- IMO, most of the time, the source code should document itself and java documentation is necessary for an API distributed as an already compiled library/jar.
- I would prefer reviewers to point to obscure code rather than to rely on the documentation.
- Waiting for a review comments helps to avoid inconsistency between the code and the documentation (quite a common problem) as usually code evolves faster and documentation lags behind. 
- I already added documentation for `ExecutableTasksLatch` and `ExecutableTask`.
- I'll change comments for `AsyncPageReader` during rebase and merge conflict resolution.

Reply:
Please be more specific. Do you refer to `AsyncPageReader` doc? It was already changed during the merge conflict resolution.